src/VisualStudio/Core/Def/Workspace/VisualStudioActiveDocumentTracker.cs

```diff
@@ -51,12 +53,12 @@ internal class VisualStudioActiveDocumentTracker : ForegroundThreadAffinitizedOb
         IThreadingContext threadingContext,
         [Import(typeof(SVsServiceProvider))] IAsyncServiceProvider asyncServiceProvider,
         IVsEditorAdaptersFactoryService editorAdaptersFactoryService)
-        : base(threadingContext, assertIsForeground: false)
     {
+        _threadingContext = threadingContext;
         _editorAdaptersFactoryService = editorAdaptersFactoryService;
-        ThreadingContext.RunWithShutdownBlockAsync(async cancellationToken =>
+        _threadingContext.RunWithShutdownBlockAsync(async cancellationToken =>
```

Comment:
❗ Asynchronous operations must have their life tracked

Reply:
I can file an issue. I'm not changing things already broken prior to this pr :-)

Reply:
https://github.com/dotnet/roslyn/issues/73048